design/design.md

```diff
@@ -0,0 +1,48 @@
+Teleport coding challenge design document by Jin Huang
+
+I) Architecture:
+The design would consist of 3 parts, an API handler, a server, and a main. Error handling will be present
+appropriately in each of the functions. To use this API, the client would request to the server with curl
+to start, stop, or query.
```

Comment:
The intern challenge you are taking does not require a HTTP server implementation. Only a worker library. Effectively the following:


```go
type worker struct {
}

func newWorker(...) (*worker, error) {
}

func (w *worker) Start() error {
}

func (w *worker) Stop(...) error {
}

func (w *worker) Status(...) (..., error) {
}
```

Reply:
Revised the design document for re-review.